contracts/System.sol

```diff
@@ -80,4 +122,171 @@ contract System {
   /// @param lowerBound requested lower bound of the param.
   /// @param upperBound requested upper bound of the param
   error OutOfBounds(string name, uint256 given, uint256 lowerBound, uint256 upperBound);
+
+  struct Addresses {
+    address validatorSet;
+    address slash;
+    address systemReward;
+    address lightClient;
+    address relayerHub;
+    address candidateHub;
+    address govHub;
+    address pledgeAgent;
+    address burn;
+    address foundation;
+  }
+
+  struct ExtStorage {
+    Addresses addrs;
+    // @dev additional extended-storage fields goes here
```

Comment:
any additional state variables added to the System must go here and not to the 'standard' sequential storage

Reply:
We'll probably need to create a guide/document on the recommended process with examples. Both for internal use and likely adding it to the README somewhere. Not required in this PR, but needed soon considering all the coming code changes with the development of various features that require code changes to the genesis contracts

Reply:
agreed. The PR description above would probably be a good starting point